std/random.d

```diff
@@ -1336,6 +1330,271 @@ version (AnyARC4Random)
     extern(C) private @nogc nothrow
     {
         uint arc4random() @safe;
+        void arc4random_buf(scope void* ptr, size_t len) @system;
+    }
+
+    /+
+    Tries to use a system API to get entropy. Resources are not cached
+    due to the expectation that this function will not be called more
+    than once per program execution.
```

Comment:
>It would be much better to name it according to what it does, e.g. something like `getSystemEntropy`.

The name is a warning to developers against using the function that way repeatedly.  The comment hints at the reason: _"Resources are not cached due to the expectation that this function will not be called more than once per program execution."_  So there may be resources that a caller would expect a reasonable implementation to reuse that are not cached for reuse here, making repeated calls more expensive than expected.

Maybe a name like `getExpensiveSystemEntropy` or `getSystemEntropyWithoutSavingResources` would be better than `oneTimeGenRandom`?

Reply:
> The name is a warning to developers against using the function that way repeatedly.

If I understand right, what you want to ensure is that users understand that this should not be used as their default random number generator.  Right?

But that's what the documentation is for.  The inefficiency that you note is an implementation detail that might be fixed later, not something inherent to the API.  It shouldn't pollute the API naming choices.

Reply:
BTW, I think it's explicitly _not_ the case that we expect "that this function will not be called more than once per program execution".  I would anticipate for example that it might be called once per thread (to seed different, thread-local pseudo-RNG instances).

Reply:
>If I understand right, what you want to ensure is that users understand that this should not be used as their default random number generator. Right?

Since it's private I was only thinking of other developers when I wrote it. I assume they know whether they want system entropy or something else. The message is: "if you want to retrieve system entropy repeatedly over the life of the application, write a different function."

>The inefficiency that you note is an implementation detail that might be fixed later

For repeated use it's an inefficiency, but for one-time use it may be optimal. We might not want to open `/dev/urandom` and keep it open for the life of the application.

Reply:
> Since it's private I was only thinking of other developers when I wrote it

That's what documentation is for (to advise developers on _how_ to use a function).

By the way, I like the way you've phrased it in this particular response ("If you want to retrieve system entropy repeatedly ...") _much_ better than how you phrased it in the actual documentation.

> For repeated use it's an inefficiency, but for one-time use it may be optimal. We might not want to open `/dev/urandom` and keep it open for the life of the application.

Oh, absolutely.  But note that this also touches on the importance of providing the different sources of entropy as entities in their own right, that can be used in multiple different ways, according to use case.

Reply:
>By the way, I like the way you've phrased it in this particular response ("If you want to retrieve system entropy repeatedly ...") much better than how you phrased it in the actual documentation.

Okay, changed it.